components/serializer.rst

```diff
@@ -922,17 +922,22 @@ The ``setCircularReferenceLimit()`` method of this normalizer sets the number
 of times it will serialize the same object before considering it a circular
 reference. Its default value is ``1``.
 
+.. deprecated:: 4.2
+
+    The :method:`Symfony\\Component\\Serializer\\Normalizer\\AbstractNormalizer::setCircularReferenceHandler`
+    method is deprecated since Symfony 4.2, use the "circular_reference_handler" key of the context instead.
+
 Instead of throwing an exception, circular references can also be handled
 by custom callables. This is especially useful when serializing entities
 having unique identifiers::
 
     $encoder = new JsonEncoder();
-    $normalizer = new ObjectNormalizer();
-
-    // all callback parameters are optional (you can omit the ones you don't use)
-    $normalizer->setCircularReferenceHandler(function ($object, string $format = null, array $context = []) {
-        return $object->getName();
-    });
+    $defaultContext = [
+            AbstractNormalizer::CIRCULAR_REFERENCE_HANDLER => function ($object, $format, $context) {
+                return $object->getName();
+            }
+        ];
```

Comment:
Little note: we should fix indentation here (the array contents should be indented with only 4 spaces and `];` should be on the same column as `$defaultContext =`). But we can do that while merging your PR.